### PR TITLE
Feat/factory contract

### LINKED ISSUE #6: Introduce `RenPoolFactory` contract to keep track of all deployed pools in a central contract
There is no tracking mechanism for an Owner or Node Operator to see their deployed `RenPool` contracts.

Introduce a new contract, `RenPoolFactory`, to keep track of all deployed pools in a central contract.

### PR DESCRIPTION
**Pull request by fede-rodes**
Closes #6 

Let think of this as a first iteration. I've been reading about Factory pattern here https://betterprogramming.pub/learn-solidity-the-factory-pattern-75d11c3e7d29 and https://consensys.net/diligence/blog/2019/09/factories-improve-smart-contract-security/

It looks like, if I understood correctly, that is possible to deploy a single instance and then use a proxy to reuse the renpool contract logic without deploying extra instances (?)

### CODE REVIEW

**Comment by acuarica** [comment]
> It looks like, if I understood correctly, that is possible to deploy a single instance and then use a proxy to reuse the renpool contract logic without deploying extra instances (?)

Not sure what you mean here by _reuse the renpool contract logic_. But sure, let's take this as first iteration.

File: test/RenPoolFactory.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        AddressZero,
+        AddressZero,
+        AddressZero,
+        AddressZero,
+        0,
+      );
+
+      const pools = await renPoolFactory.getPools();
+
+      expect(pools.length).to.equal(1);
```
Comment (acuarica) [inline test/RenPoolFactory.test.js:58]: Should we add some sort of smoke test on the deployed contract? In the sense, how are we making sure that the actual deployment works?

**Review by acuarica — APPROVED** [review]
lgtm

File: test/RenPoolFactory.test.js (hunk 1 above)
Comment (fede-rodes) [reply test/RenPoolFactory.test.js:58]: agree, I'll add it!

**Review by acuarica — APPROVED** [review]
lgtm

**Pull request merged**